objectbox/example/flutter/objectbox_demo/lib/objectbox.dart

```diff
@@ -0,0 +1,51 @@
+import 'model.dart';
+import 'objectbox.g.dart'; // created by `flutter pub run build_runner build`
+
+/// Provides access to the ObjectBox Store throughout the app.
+///
+/// Initialized in the apps main function.
+class ObjectBox {
+  Store? _store;
+
+  /// A Box of notes.
+  late final Box<Note> noteBox;
```

Comment:
As an alternative to my suggestion above, this "wrapper" could be part of the generated code but I'd suggest using an async factory instead of constructing an incomplete object and then having to call `init()`.

Reply:
Makes sense as at least right now `init()` does not have any parameters. My initial thought was to prevent any accidental initialization from any other place than the `main()` function, so separated actual init from the getter and made the getter throw.

Do you maybe have a short snippet? I guess this should work using this `??=` syntax? Somewhat like https://github.com/alextekartik/flutter_app_example/blob/master/notepad_sqflite/lib/provider/note_provider.dart#L60-L65